File: scala/src/test/scala/com/gildedrose/GildedRoseTest.scala
package com.gildedrose

import java.nio.file.{Files, Paths}

import org.scalatest._

class GildedRoseTest  extends FlatSpec with Matchers {
      it should "foo" in {
        val expected = new String(Files.readAllBytes(Paths.get(this.getClass.getResource("golden-master.txt").toURI)))
        val result = new GildedRoseExecutor().execute(50)
        (expected) should equal (result)
      }
}

File: Java/src/test/java/com/gildedrose/GildedRoseTest.java
package com.gildedrose;

import static org.junit.Assert.*;

import org.junit.Test;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.PrintStream;
import java.net.URISyntaxException;
import java.net.URL;
import java.nio.file.Files;
import java.nio.file.Paths;

public class GildedRoseTest {

    @Test
    public void testGoldenMaster() throws URISyntaxException, IOException {
        String expected = new String(Files.readAllBytes(Paths.get(this.getClass().getResource("golden-master.txt").toURI())));
        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
             PrintStream printStream = new PrintStream(byteArrayOutputStream)) {
            System.setOut(printStream);
            TexttestFixture.main(new String[]{"50"});
            String result = byteArrayOutputStream.toString();
            assertEquals(expected, result);
        }
    }

}
